### PR TITLE
Add etcd integrations tests

### PR DESCRIPTION
**Pull request by mrjana**
- Added etcd integration test for overlay
    - Added etcd integration test for multinode
      with mock test driver suitable for circleci
    - Added multinode tests for zookeeper
    - Made the script smart enough to only start
      data stores necessary for the requested suites

Signed-off-by: Jana Radhakrishnan <mrjana@docker.com>

### CODE REVIEW

**Comment by chenchun** [comment]
Looks good than my PR. Closed mine.

**Comment by abronan** [comment]
Maybe a small nit on the port choice (why 42000 and not 4001?).

But LGTM. Great to have tests for the 3 stores now.

**Comment by tomdee** [comment]
Yes big :+1: to having tests for all three stores. Thanks for getting these tests in.

File: test/integration/dnet/helpers.bash [hunk 1]
```diff
@@ -207,6 +213,27 @@ function runc() {
...
 
+function start_etcd() {
+    local bridge_ip
+    stop_etcd
+
+    bridge_ip=$(get_docker_bridge_ip)
+    docker run -d \
+	   --net=host \
+	   --name=dn_etcd \
+	   mrjana/etcd --listen-client-urls http://0.0.0.0:42000 \
```
Comment (aboch) [inline test/integration/dnet/helpers.bash]: `mrjana/etcd` ?

**Comment by mrjana** [comment]
@abronan I chose 42000 since 4001 is a common etcd port which folks might be using in their dev machine (if they are playing with etcd separately) and we start the etcd container in host namespace. So just to avoid that case

**Comment by mavenugo** [comment]
thanks @mrjana . huge :+1: from me. LGTM.

**Comment by abronan** [comment]
@mrjana Ok makes sense ;) then you might probably want to change the consul port to something else than `8500` to make sure it does not end up to the state you described.

**Comment by mrjana** [comment]
@abronan Yeah that makes sense too but since I start consul in bridge network somehow didn't pay attention to the port but we are really mapping container port 8500 to host port 8500. Will do that in another PR. Thanks :-)

**Pull request merged**